### PR TITLE
build-configs-cip.yaml: Add some blocklists for the 4.4 series

### PR DESCRIPTION
**Pull request by fbezdeka**
**Note: Not ready for merging yet, let's wait for some test results**

Trying to silence some build failures for invalid
kernel 4.4 - compiler - kernel configuration combinations. That should
help us to separate "noise" from real issues.

Signed-off-by: Florian Bezdeka <florian.bezdeka@siemens.com>

### CODE REVIEW

**Comment by fbezdeka** [comment]
@aliceinwire could you have a look?

It would be interesting to find a 4.4-stable log somewhere but I guess it was garbage collected since 4.4-stable is EOL.

Any chance to test that? According to documentation pull requests should be tested automatically on staging, but I was not able to find some (build-) logs somewhere.

**Comment by aliceinwire** [comment]
the test is automatically done in staging.kernelci.org
the code looks good at first review.
The cip logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.

**Review by aliceinwire — APPROVED** [review]

**Comment by aliceinwire** [comment]
did you run validate on the scripts for validating the code ?

**Comment by aliceinwire** [comment]
```
2022-07-19T10:11:15.4761830Z E           yaml.scanner.ScannerError: while scanning an anchor
2022-07-19T10:11:15.4762270Z E             in "config/core/build-configs-cip.yaml", line 94, column 19
2022-07-19T10:11:15.4762707Z E           expected alphabetic or numeric character, but found '.'
2022-07-19T10:11:15.4763209Z E             in "config/core/build-configs-cip.yaml", line 94, column 34
```

**Comment by aliceinwire** [comment]
that doesn't looks like the way of writing yml files

File: config/core/build-configs-cip.yaml [hunk 1]
```diff
@@ -91,22 +91,56 @@ cip_variants_preempt_rt: &cip_variants_preempt_rt
           - 'x86_64_defconfig+preempt_rt'
           - 'x86_64_defconfig+x86-chromebook+preempt_rt'
 
+cip_variants_4.4: &cip_variants_4.4:
```
Comment (aliceinwire) [inline config/core/build-configs-cip.yaml:94]:
```
2022-07-19T10:11:15.4761830Z E           yaml.scanner.ScannerError: while scanning an anchor
2022-07-19T10:11:15.4762270Z E             in "config/core/build-configs-cip.yaml", line 94, column 19
2022-07-19T10:11:15.4762707Z E           expected alphabetic or numeric character, but found '.'
2022-07-19T10:11:15.4763209Z E             in "config/core/build-configs-cip.yaml", line 94, column 34
```

**Review by aliceinwire — CHANGES_REQUESTED** [review]

**Comment by fbezdeka** [comment]
> The CIP logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.

Nice, but in addition it would be nice to compare the 4.4-cip logs to the 4.4 (stable) logs. I would like to now if there is a problem in the 4.4-cip tree or not. When looking at the 4.4 configs here I can't find such exclusions, so I'm aksing myself why this is necessary at all.

File: config/core/build-configs-cip.yaml (hunk 1 above)
Comment (fbezdeka) [reply config/core/build-configs-cip.yaml]: My fault. Regex gone wild (after validation...)

**Comment by fbezdeka** [comment]
> did you run validate on the scripts for validating the code ? and run `python3 -m pytest tests` ? Please run both

There is still one problem with the tests, but I'm not able to fix it. Something with the build_environment seems wrong.

**Comment by fbezdeka** [comment]
> > > The CIP logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.
> > 
> > 
> > Nice, but in addition it would be nice to compare the 4.4-cip logs to the 4.4 (stable) logs. I would like to now if there is a problem in the 4.4-cip tree or not. When looking at the 4.4 configs here I can't find such exclusions, so I'm aksing myself why this is necessary at all.
> 
> That's something that is not implemented yet. we can do regression but not out of branch regression AFAIK probably will be implemented with the new KernelCI API

I'm just searching for the last 4.4-stable log. No "compare feature" requested here... I assume that the last run for the 4.4-stable series was successful but the CIP runs are not. So the question is why? There must be a difference. If this difference is inside the 4.4-cip tree we have a real regression and should not silence the tests.

**Review by aliceinwire — APPROVED** [review]

**Comment by aliceinwire** [comment]
Please remove the draft status so that we can test the pull request in staging kernelci

**Comment by fbezdeka** [comment]
> 



> > > > > The CIP logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.
> > > > 
> > > > 
> > > > Nice, but in addition it would be nice to compare the 4.4-cip logs to the 4.4 (stable) logs. I would like to now if there is a problem in the 4.4-cip tree or not. When looking at the 4.4 configs here I can't find such exclusions, so I'm aksing myself why this is necessary at all.
> > > 
> > > 
> > > That's something that is not implemented yet. we can do regression but not out of branch regression AFAIK probably will be implemented with the new KernelCI API
> > 
> > 
> > I'm just searching for the last 4.4-stable log. No "compare feature" requested here... I assume that the last run for the 4.4-stable series was successful but the CIP runs are not. So the question is why? There must be a difference. If this difference is inside the 4.4-cip tree we have a real regression and should not silence the tests.
> 
> until now the garbage collector was just keeping 28 days of logs.from now we can keep logs for more time.

Nice. Could we somehow trigger a rebuild of the EOL 4.4-stable tree? That would allow me to compare the results.

**Comment by aliceinwire** [comment]
> 



> > 
> 
> > > > > > The CIP logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.
> > > > > 
> > > > > 
> > > > > Nice, but in addition it would be nice to compare the 4.4-cip logs to the 4.4 (stable) logs. I would like to now if there is a problem in the 4.4-cip tree or not. When looking at the 4.4 configs here I can't find such exclusions, so I'm aksing myself why this is necessary at all.
> > > > 
> > > > 
> > > > That's something that is not implemented yet. we can do regression but not out of branch regression AFAIK probably will be implemented with the new KernelCI API
> > > 
> > > 
> > > I'm just searching for the last 4.4-stable log. No "compare feature" requested here... I assume that the last run for the 4.4-stable series was successful but the CIP runs are not. So the question is why? There must be a difference. If this difference is inside the 4.4-cip tree we have a real regression and should not silence the tests.
> > 
> > 
> > until now the garbage collector was just keeping 28 days of logs.from now we can keep logs for more time.
> 
> Nice. Could we somehow trigger a rebuild of the EOL 4.4-stable tree? That would allow me to compare the results.

ok

**Comment by aliceinwire** [comment]
stable 4.4 rebuild 

> > 
> 
> > > 
> > 
> > 
> > > > > > > The CIP logs garbage collector as been excluded for CIP logs so we will be able to keep older logs in the future.
> > > > > > 
> > > > > > 
> > > > > > Nice, but in addition it would be nice to compare the 4.4-cip logs to the 4.4 (stable) logs. I would like to now if there is a problem in the 4.4-cip tree or not. When looking at the 4.4 configs here I can't find such exclusions, so I'm aksing myself why this is necessary at all.
> > > > > 
> > > > > 
> > > > > That's something that is not implemented yet. we can do regression but not out of branch regression AFAIK probably will be implemented with the new KernelCI API
> > > > 
> > > > 
> > > > I'm just searching for the last 4.4-stable log. No "compare feature" requested here... I assume that the last run for the 4.4-stable series was successful but the CIP runs are not. So the question is why? There must be a difference. If this difference is inside the 4.4-cip tree we have a real regression and should not silence the tests.
> > > 
> > > 
> > > until now the garbage collector was just keeping 28 days of logs.from now we can keep logs for more time.
> > 
> > 
> > Nice. Could we somehow trigger a rebuild of the EOL 4.4-stable tree? That would allow me to compare the results.
> 
> currently staging KernelCI is under maintenance. will post here the results when finished.

I just pushed the rebuild of EOL 4.4 stable tree
https://staging.kernelci.org/job/stable/branch/linux-4.4.y/

**Comment by gctucker** [comment]
> Please remove the draft status so that we can test the pull request in staging kernelci

Draft PRs are still deployed on staging. This one hasn't been deployed yet because @fbezdeka was not on the list of "trusted" GitHub users. Now that's been addressed, so the next staging branch should include this PR.

**Comment by gctucker** [comment]
FYI this is how it's done: https://github.com/kernelci/kernelci-deploy/commit/31a11b4f96f99dcf61b1083765b4bc1df58a4f8f

**Comment by aliceinwire** [comment]
> FYI this is how it's done: [kernelci/kernelci-deploy@31a11b4](https://github.com/kernelci/kernelci-deploy/commit/31a11b4f96f99dcf61b1083765b4bc1df58a4f8f)

thanks

**Comment by fbezdeka** [comment]
Thank you all!

My interpretation is now that 4.4-stable had the same issues and nobody cared... Right?

**Comment by fbezdeka** [comment]
Are there some results from this PR available somewhere? I can't find them on staging...

**Comment by aliceinwire** [comment]
I give a start to cip_4.4 with the staging code https://staging.kernelci.org/test/job/cip/branch/linux-4.4.y-cip/kernel/v4.4.302-cip69-517-g92709d8ae585/

**Comment by aliceinwire** [comment]
The errors correctly disappeared.

**Comment by aliceinwire** [comment]
This are the results differences:
https://gist.github.com/aliceinwire/251b324fee1ce965d51db0b72f5dd011/revisions?diff=unified#diff-a92e01a033c084e6b1364a6737dd4c25de3a2905616d4e97b4b418d4fb67ab9b

**Comment by fbezdeka** [comment]
OK, so no build failures anymore (great!) but some test failures:

- NFS: Seems a broken network config in the lab (DHCP not working?, @aliceinwire who could fix that?)
- SMC: Lists some CVEs
- sleep: Seems a problem with RTC on chromebooks

Maybe something we could/should discuss at the next CIP meeting together with @jan-kiszka.

**Pull request merged**